src/SciGlassCalorimeterTiltless_geo.cpp

```diff
@@ -0,0 +1,376 @@
+    wedge_box_side_v.setVisAttributes(lcdd.visAttributes(wedge_box_handle.visStr()));
+
+    int sector = 0;
+    double sector_phi = sectors_handle.phi0();
+    for (; sector < sectors_handle.number();
+         sector++, sector_phi += sectors_handle.deltaphi()) {
+      envelope_v.placeVolume(
+        wedge_box_side_v,
+        Transform3D{RotationZ{sector_phi + sectors_handle.deltaphi() / 2}}
+        );
+    }
+
+    // TODO: The endcap sides of the box are not implemented
+  }
+
+  int sector = 0;
+  double sector_phi = sectors_handle.phi0();
+  for (; sector < sectors_handle.number();
+       sector++, sector_phi += sectors_handle.deltaphi()) {
+    int row = 0;
+    double row_phi = -rows_handle.deltaphi() / 2 * (rows_handle.number() - 1);
+    for (; row < rows_handle.number();
+         row++, row_phi += rows_handle.deltaphi()) {
+
+      const double tower_gap_longitudinal = dim_handle.gap();
+
+      // negative rapidity towers will be counted backwards from -1
+      std::array<int, 2> tower_ids = {-1, 0};
+      std::array<double, 2> betas = {0., 0.};
+      std::array<double, 2> beta_prevs = {0., 0.};
+      std::array<double, 2> dzs = {0., 0.};
+      std::array<double, 2> flare_angle_polar_prevs = {0., 0.};
+
+      for (xml_coll_t family_handle{rows_handle, _Unicode(family)};
+           family_handle; ++family_handle) {
+        const int dir_sign = family_handle.attr<double>(_Unicode(dir_sign));
+        int &tower_id = tower_ids[(dir_sign > 0) ? 1 : 0];
+        double &beta = betas[(dir_sign > 0) ? 1 : 0];
+        double &beta_prev = beta_prevs[(dir_sign > 0) ? 1 : 0];
+        double &dz = dzs[(dir_sign > 0) ? 1 : 0];
+        double &flare_angle_polar_prev =
+            flare_angle_polar_prevs[(dir_sign > 0) ? 1 : 0];
+
+        xml_dim_t family_dim_handle = family_handle;
+        const double length = family_dim_handle.z_length();
+        const auto flare_angle_polar =
+            family_dim_handle.attr<double>(_Unicode(flare_angle_polar));
+        const unsigned int number = family_dim_handle.number();
+        const auto flare_angle_at_face =
+            family_dim_handle.attr<double>(_Unicode(flare_angle_at_face));
+
+        const double z = length / 2;
+        const double y1 = family_dim_handle.y1();
+        const double y2 = y1 + length * tan(flare_angle_polar);
+        const double x1 =
+            family_dim_handle.x1() +
+            (dir_sign < 0) * (2 * y1) * tan(flare_angle_at_face);
+        const double x2 =
+            family_dim_handle.x1() +
+            (dir_sign > 0) * (2 * y1) * tan(flare_angle_at_face);
+        const double x3 = x1 * (y2 / y1);
+        const double x4 = x2 * (y2 / y1);
+        const double theta = 0.;
+        const double phi = 0.;
+        const double alpha1 = 0.;
+        const double alpha2 = 0.;
+
+        for (unsigned int tower = 0; tower < number; tower++, tower_id += dir_sign) {
```

Comment:
Is it possible to create a single module before the loop, and then just place it in the loop? This would allow geant4 to optimize the geometry better and treat tracks going into any block in the same row as if they are going into the same block.

Reply:
I went through this exercise once before. And even though the code looked much better, I saw a significant performance hit in Geant4 simulation. Back then I reverted without investigation. This would be nice to have implemented sometime.

Reply:
Ok, we'll put it on the TODO list :-)